contracts/stellar-interchain-token-service/src/contract.rs

```diff
@@ -298,7 +298,7 @@ impl InterchainTokenServiceInterface for InterchainTokenService {
             destination_chain: destination_chain.clone(),
             destination_address: destination_address.clone(),
             amount,
-            data: data.clone(),
+            data_hash: data.as_ref().map(|d| env.crypto().keccak256(d).into()),
```

Comment:
```suggestion
            data_hash: data.as_ref().map(|data| env.crypto().keccak256(data).into()),
```
Use meaningful names. It's a common pattern to shadow the variable itself when unwrapping etc.

Reply:
updated this line and `InterchainTransferReceivedEvent` as well